src/engine/source/api/src/test/handlers.cpp

```diff
@@ -458,6 +464,117 @@ api::Handler sessionsDelete(shared_ptr<Router> router, shared_ptr<Catalog> catal
     };
 }
 
+api::Handler runPost(std::shared_ptr<::router::Router> router)
+{
+    return [router](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::RunPost_Request;
+        using ResponseType = eTest::RunPost_Response;
+        auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        auto& eRequest = std::get<RequestType>(res);
+        auto errorMsg = !eRequest.has_name()    ? std::make_optional("Missing /session name")
+                        : !eRequest.has_event() ? std::make_optional("Missing /event")
+                                                : std::nullopt;
+
+        if (!eRequest.has_debugmode())
+        {
+            eRequest.set_debugmode(eTest::DebugMode::ONLY_OUTPUT);
+        }
+
+        if (!eRequest.has_protocolqueue())
+        {
+            eRequest.set_protocolqueue(TEST_DEFAULT_PROTOCOL_QUEUE);
+        }
+
+        if (!eRequest.has_protocollocation())
+        {
+            eRequest.set_protocollocation(TEST_DEFAULT_PROTOCOL_LOCATION);
+        }
```

Comment:
Wouldn't it be better to set the values on local variables instead of using the request object?